### PR TITLE
[CHORE] Setup foundation classes for supporting GraphQL

### LINKED ISSUE #14: ✨ **[TASK-3] As a user, I can see sign-in screen and use my email and pasword to sign in** ✨
**Descriptions**
When first entering the app, users will see the Login screen. When signing-in, they will be redirected to the home dashboard.

**Design**
[Login UI](https://www.figma.com/proto/GjRPOjDyZ6f4EDL3wKarRK/Challenge---Mobile-App?node-id=31%3A195&scaling=scale-down)

**Acceptance Criteria**
- [x] Implement UI for login
- [x] Connect to login API

PR: #2, #3, #5, #8

### PR DESCRIPTION
**Pull request by minhnimble**
## What happened

Creating foundation classes for for supporting GraphQL request.
 
## Insight

- Added the Apollo library using Pod that helps handling GraphQL on iOS.
- Added a sample query `LaunchList.graphql` for testing purpose.
- Connected with sample endpoint `https://apollo-fullstack-tutorial.herokuapp.com/` for testing purpose.
- Added supported code for future user's session management.
- Added foundation classes for handling RESTful on iOS.
 
## Proof Of Work

A sample request and response with GraphQL is triggered successfully when the app starts with internet connection should appear in the console log:
```
2020-12-03 17:43:52.729043+0700 Reportedly-Staging[49607:514680] [Logger] RequestLoggingInterceptor.interceptAsync(chain:request:response:completion:):19
> 🍏 Outgoing request: HTTPRequest details:
	Endpoint: https://apollo-fullstack-tutorial.herokuapp.com/
	Additional Headers: [
		X-APOLLO-OPERATION-NAME: LaunchList,
		apollographql-client-name: nimble.Reportedly.staging-apollo-ios,
		apollographql-client-version: 1.0-1,
		Content-Type: application/json,
		X-APOLLO-OPERATION-TYPE: query,
	]
	Cache Policy: returnCacheDataElseFetch
	Operation: Reportedly_Staging.LaunchListQuery
	Context identifier: nil
2020-12-03 17:43:52.970586+0700 Reportedly-Staging[49607:514680] [] nw_protocol_get_quic_image_block_invoke dlopen libquic failed
2020-12-03 17:43:54.640704+0700 Reportedly-Staging[49607:514583] [Logger] ResponseLoggingInterceptor.interceptAsync(chain:request:response:completion:):38
> 🍏 HTTP Response: <NSHTTPURLResponse: 0x6000026699a0> { URL: https://apollo-fullstack-tutorial.herokuapp.com/ } { Status Code: 200, Headers {
    "Access-Control-Allow-Origin" =     (
        "*"
    );
    Connection =     (
        "keep-alive"
    );
    "Content-Length" =     (
        1219
    );
    "Content-Type" =     (
        "application/json; charset=utf-8"
    );
    Date =     (
        "Thu, 03 Dec 2020 10:43:54 GMT"
    );
    Etag =     (
        "W/\"4c3-o2UuHmQO54uxxukxc7YQudROZEU\""
    );
    Server =     (
        Cowboy
    );
    Via =     (
        "1.1 vegur"
    );
} }
2020-12-03 17:43:54.641991+0700 Reportedly-Staging[49607:514583] [Logger] ResponseLoggingInterceptor.interceptAsync(chain:request:response:completion:):41
> 🍏 Data: {"data":{"launches":{"__typename":"LaunchConnection","cursor":"1581951955","hasMore":true,"launches":[{"__typename":"Launch","id":"109","site":"CCAFS SLC 40"},{"__typename":"Launch","id":"108","site":"VAFB SLC 4E"},{"__typename":"Launch","id":"107","site":"KSC LC 39A"},{"__typename":"Launch","id":"106","site":"CCAFS SLC 40"},{"__typename":"Launch","id":"105","site":"CCAFS SLC 40"},{"__typename":"Launch","id":"104","site":"KSC LC 39A"},{"__typename":"Launch","id":"103","site":"KSC LC 39A"},{"__typename":"Launch","id":"102","site":"KSC LC 39A"},{"__typename":"Launch","id":"101","site":"CCAFS SLC 40"},{"__typename":"Launch","id":"100","site":"CCAFS SLC 40"},{"__typename":"Launch","id":"99","site":"KSC LC 39A"},{"__typename":"Launch","id":"98","site":"CCAFS SLC 40"},{"__typename":"Launch","id":"97","site":"CCAFS SLC 40"},{"__typename":"Launch","id":"96","site":"CCAFS SLC 40"},{"__typename":"Launch","id":"95","site":"CCAFS SLC 40"},{"__typename":"Launch","id":"94","site":"KSC LC 39A"},{"__typename":"Launch","id":"93","site":"KSC LC 39A"},{"__typename":"Launch","id":"92","site":"KSC LC 39A"},{"__typename":"Launch","id":"91","site":"CCAFS SLC 40"},{"__typename":"Launch","id":"90","site":"CCAFS SLC 40"}]}}}
Success! Result: GraphQLResult<Data>(data: Optional(Reportedly_Staging.LaunchListQuery.Data(resultMap: ["launches": Optional(["__typename": Optional("LaunchConnection"), "cursor": Optional("1581951955"), "hasMore": Optional(true), "launches": Optional([Optional(["id": Optional("109"), "__typename": Optional("Launch"), "site": Optional("CCAFS SLC 40")]), Optional(["site": Optional("VAFB SLC 4E"), "__typename": Optional("Launch"), "id": Optional("108")]), Optional(["__typename": Optional("Launch"), "id": Optional("107"), "site": Optional("KSC LC 39A")]), Optional(["__typename": Optional("Launch"), "id": Optional("106"), "site": Optional("CCAFS SLC 40")]), Optional(["id": Optional("105"), "site": Optional("CCAFS SLC 40"), "__typename": Optional("Launch")]), Optional(["site": Optional("KSC LC 39A"), "__typename": Optional("Launch"), "id": Optional("104")]), Optional(["__typename": Optional("Launch"), "id": Optional("103"), "site": Optional("KSC LC 39A")]), Optional(["__typename": Optional("Launch"), "id": Optional("102"), "site": Optional("KSC LC 39A")]), Optional(["__typename": Optional("Launch"), "site": Optional("CCAFS SLC 40"), "id": Optional("101")]), Optional(["site": Optional("CCAFS SLC 40"), "id": Optional("100"), "__typename": Optional("Launch")]), Optional(["id": Optional("99"), "__typename": Optional("Launch"), "site": Optional("KSC LC 39A")]), Optional(["site": Optional("CCAFS SLC 40"), "__typename": Optional("Launch"), "id": Optional("98")]), Optional(["site": Optional("CCAFS SLC 40"), "id": Optional("97"), "__typename": Optional("Launch")]), Optional(["id": Optional("96"), "site": Optional("CCAFS SLC 40"), "__typename": Optional("Launch")]), Optional(["id": Optional("95"), "site": Optional("CCAFS SLC 40"), "__typename": Optional("Launch")]), Optional(["site": Optional("KSC LC 39A"), "__typename": Optional("Launch"), "id": Optional("94")]), Optional(["id": Optional("93"), "__typename": Optional("Launch"), "site": Optional("KSC LC 39A")]), Optional(["__typename": Optional("Launch"), "id": Optional("92"), "site": Optional("KSC LC 39A")]), Optional(["__typename": Optional("Launch"), "site": Optional("CCAFS SLC 40"), "id": Optional("91")]), Optional(["id": Optional("90"), "site": Optional("CCAFS SLC 40"), "__typename": Optional("Launch")])])])])), errors: nil, extensions: nil, source: Apollo.GraphQLResult<Reportedly_Staging.LaunchListQuery.Data>.Source.server, dependentKeys: Optional(Set(["QUERY_ROOT.launches.launches.9.id", "QUERY_ROOT.launches.launches.19.id", "QUERY_ROOT.launches.launches.12.id", "QUERY_ROOT.launches.launches.18.site", "QUERY_ROOT.launches.launches.13.site", "QUERY_ROOT.launches.launches.12.site", "QUERY_ROOT.launches.launches.4.site", "QUERY_ROOT.launches.launches.14.__typename", "QUERY_ROOT.launches.launches.3.__typename", "QUERY_ROOT.launches.launches.2.id", "QUERY_ROOT.launches.launches.12.__typename", "QUERY_ROOT.launches.launches.5.site", "QUERY_ROOT.launches.launches.11.__typename", "QUERY_ROOT.launches.launches.8.id", "QUERY_ROOT.launches.launches.2.__typename", "QUERY_ROOT.launches.launches.19.site", "QUERY_ROOT.launches.launches", "QUERY_ROOT.launches.launches.10.id", "QUERY_ROOT.launches.launches.13.id", "QUERY_ROOT.launches.launches.0.site", "QUERY_ROOT.launches.launches.3.id", "QUERY_ROOT.launches.cursor", "QUERY_ROOT.launches.launches.0.id", "QUERY_ROOT.launches.launches.8.site", "QUERY_ROOT.launches.launches.6.id", "QUERY_ROOT.launches.launches.7.__typename", "QUERY_ROOT.launches.launches.6.site", "QUERY_ROOT.launches.launches.11.site", "QUERY_ROOT.launches.launches.14.site", "QUERY_ROOT.launches.launches.8.__typename", "QUERY_ROOT.launches.launches.0.__typename", "QUERY_ROOT.launches.launches.15.id", "QUERY_ROOT.launches.launches.17.__typename", "QUERY_ROOT.launches.launches.17.site", "QUERY_ROOT.launches.launches.11.id", "QUERY_ROOT.launches.launches.10.__typename", "QUERY_ROOT.launches.launches.16.site", "QUERY_ROOT.launches.__typename", "QUERY_ROOT.launches.launches.18.__typename", "QUERY_ROOT.launches.launches.18.id", "QUERY_ROOT.launches.launches.6.__typename", "QUERY_ROOT.launches.launches.19.__typename", "QUERY_ROOT.launches.launches.15.site", "QUERY_ROOT.launches.launches.7.id", "QUERY_ROOT.launches.launches.9.site", "QUERY_ROOT.launches.launches.2.site", "QUERY_ROOT.launches.launches.10.site", "QUERY_ROOT.launches.launches.4.__typename", "QUERY_ROOT.launches.launches.4.id", "QUERY_ROOT.launches.launches.9.__typename", "QUERY_ROOT.launches.launches.1.__typename", "QUERY_ROOT.launches.launches.7.site", "QUERY_ROOT.launches.launches.1.id", "QUERY_ROOT.launches.launches.5.id", "QUERY_ROOT.launches.launches.16.id", "QUERY_ROOT.launches.hasMore", "QUERY_ROOT.launches.launches.16.__typename", "QUERY_ROOT.launches.launches.14.id", "QUERY_ROOT.launches.launches.15.__typename", "QUERY_ROOT.launches.launches.17.id", "QUERY_ROOT.launches.launches.1.site", "QUERY_ROOT.launches", "QUERY_ROOT.launches.launches.3.site", "QUERY_ROOT.launches.launches.5.__typename", "QUERY_ROOT.launches.launches.13.__typename"])))
```

### CODE REVIEW

**Review by tamv263 — COMMENTED** [review]
Everything looks great, i have just a few comments 😄

File: Podfile [hunk 1]
```diff
@@ -6,11 +6,12 @@ target 'Reportedly' do
   use_frameworks!
 
   # Pods for Reportedly
+  pod 'Alamofire'
```
[the remark this answers is not in the document]
Comment (minhnimble) [reply Podfile:9]: yeah sure, let me remove 👍

File: Reportedly/Services/Network/Network.swift [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+//  Network.swift
+//  Reportedly
+//
+//  Created by Mikey Pham on 12/3/20.
+//  Copyright © 2020 NimbleHQ. All rights reserved.
+//
+
+import Apollo
+
+class Network {
```
[the remark this answers is not in the document]
Comment (minhnimble) [reply Reportedly/Services/Network/Network.swift:11]: noted 👍

File: Reportedly/Services/Network/UserManagementInterceptor.swift [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+//  UserManagementInterceptor.swift
+//  Reportedly
+//
+//  Created by Mikey Pham on 12/3/20.
+//  Copyright © 2020 NimbleHQ. All rights reserved.
+//
+
+import Apollo
+
+class UserManagementInterceptor: ApolloInterceptor {
```
[the remark this answers is not in the document]
Comment (minhnimble) [reply Reportedly/Services/Network/UserManagementInterceptor.swift:11]: yeah sure, let me update 👍

**Review by tamv263 — APPROVED** [review]
LGTM!

**Review by AVI5HEK — APPROVED** [review]

**Pull request merged**